src/main/java/ev3dev/utils/DataChannelRereader.java

```diff
@@ -50,15 +50,14 @@ public DataChannelRereader(String pathString) {
      */
     public String readString() {
         try {
-            int n;
-            do {
-                byteBuffer.clear();
-                channel.position(0);
-                n = channel.read(byteBuffer);
-                if (n == -1) {
-                    throw new IOException("Premature end of file ");
-                }
-            } while (n <= 0);
+            byteBuffer.clear();
+            channel.position(0);
+            int n = channel.read(byteBuffer);
```

Comment:
The `position()` + `read()` calls can be merged together: https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/1a18470748c1fca44afd342acb37c4335e654774/src/main/java/ev3dev/utils/DataChannelRereader.java#L54 With that change, all FileChannel accesses should be thread-safe (I think including close), only the buffer would be unsafe.